packages/settings/src/settings.ts

```diff
@@ -0,0 +1,56 @@
+import deepFreezeStrict from "deep-freeze-strict";
+import { Settings } from "../api";
+
+// These properties are defined (with their default values) in the package.json of the client
+const defaultSettings: Settings = {
+  codeAssist: { deprecated: false, experimental: false },
+};
+deepFreezeStrict(defaultSettings);
+
+// The global settings, used when the `workspace/configuration` request is not supported by the client.
```

Comment:
Does this mean we only support a per document configuration?
Can the user define that for ALL documents they want experimental?

Reply:
It can be defined on user or workspace level
![image](https://user-images.githubusercontent.com/13255544/83967411-b19d7280-a8c9-11ea-8596-1c921e437086.png)
On theia it looks like it can be defined per folder or project
![image](https://user-images.githubusercontent.com/13255544/83967432-e7425b80-a8c9-11ea-83f1-4ec5c0039cea.png)
Whatever the user didn't set is not written to any file and the client falls back to the more general setting.